fastcrypto-zkp/src/bn254/zk_login.rs

```diff
@@ -1,86 +1,144 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use ark_crypto_primitives::snark::SNARK;
-use std::collections::HashMap;
-use std::fmt;
-
-use super::{poseidon::PoseidonWrapper, verifier::process_vk_special};
-use crate::bn254::VerifyingKey as Bn254VerifyingKey;
-use crate::circom::CircomPublicInputs;
-use crate::{
-    bn254::verifier::PreparedVerifyingKey,
-    circom::{g1_affine_from_str_projective, g2_affine_from_str_projective},
-};
+use fastcrypto::error::FastCryptoResult;
+use serde_json::Value;
+
+use super::poseidon::PoseidonWrapper;
+use crate::circom::{g1_affine_from_str_projective, g2_affine_from_str_projective};
 pub use ark_bn254::{Bn254, Fr as Bn254Fr};
 pub use ark_ff::ToConstraintField;
-use ark_groth16::{Groth16, Proof, VerifyingKey};
+use ark_groth16::Proof;
 pub use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
 use fastcrypto::{
     error::FastCryptoError,
     rsa::{Base64UrlUnpadded, Encoding},
 };
-use num_bigint::{BigInt, BigUint};
-use once_cell::sync::Lazy;
-use regex::Regex;
+use num_bigint::{BigInt, BigUint, Sign};
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
-use serde_json::Value;
 use std::str::FromStr;
 
+type ParsedJWKs = Vec<((String, String), OAuthProviderContent)>;
```

Comment:
Seems like there is some redundancy here.. E.g., `kid` is both inside `OAuthProviderContent` and in the first tuple. Can this be avoided?

Reply:
removed